src/gui/tracks/TrackOperationsWidget.cpp

```diff
@@ -348,6 +350,16 @@ void TrackOperationsWidget::updateMenu()
 		toMenu->addAction( tr( "Turn all recording off" ), this, SLOT(recordingOff()));
 	}
 
+	PatternTrackView* patternTrackView = dynamic_cast<PatternTrackView *>(m_trackView);
+	if (patternTrackView)
+	{
+		int patternIndex = patternTrackView->getPatternTrack()->patternIndex();
+		toMenu->addSeparator();
+		toMenu->addAction(tr("Convert PatternTrack to Song Editor"), [patternIndex](){
```

Comment:
```
](){
```
Block braces on separate line.
```
]()
{
```

Reply:
I personally find this style doesn't work too well with lambdas. If you don't indent the braces, you get misleading indentation, where it's not obvious that the block is part of the statement on the preceding line:
```cpp
toMenu->addAction(tr("Convert PatternTrack to Song Editor"), [patternIndex]()
{
	Engine::getSong()->convertPatterntoSE(true, patternIndex);
}
);
```
If you do indent them, then the lambda body ends up double-indented, which starts to eat up horizontal space and compounds quite quickly when you have nested lambdas:
```cpp
toMenu->addAction(tr("Convert PatternTrack to Song Editor"), [patternIndex]()
	{
		Engine::getSong()->convertPatterntoSE(true, patternIndex);
	}
);
```
Also, it feels odd to have such separation between the lambda introduction/header (I'm not sure what the technical term is) and the opening brace, which is something you don't get with a normal function definition. That could be addressed with something like the following:
```cpp
toMenu->addAction(tr("Convert PatternTrack to Song Editor"),
	[patternIndex]()
	{
		Engine::getSong()->convertPatterntoSE(true, patternIndex);
	}
);
```
(It's a discussion better suited for elsewhere, but I don't like putting braces on their own lines in general within function bodies. I think it wastes vertical space for little increase in clarity, and having code actually visible on the screen is the number-one contribution to readability. But this is one place where I think it doesn't work in particular, for the additional reasons listed above.)